src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -228,22 +262,12 @@ async ValueTask DecrementInFlightCountAsync(InFlightSolution inFlightSolution)
         {
             try
             {
-                // Try to create the solution snapshot incrementally off of the workspaces CurrentSolution first.
-                var updater = new SolutionCreator(Services.HostServices, assetProvider, this.CurrentSolution);
-                if (await updater.IsIncrementalUpdateAsync(solutionChecksum, cancellationToken).ConfigureAwait(false))
-                {
-                    return await updater.CreateSolutionAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
-                }
-                else
-                {
-                    // Otherwise, this is a different solution, or the first time we're creating this solution.  Bulk
-                    // sync over all assets for it.
-                    await assetProvider.SynchronizeSolutionAssetsAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
+                var solutionToUpdate = await GetOrCreateSolutionToUpdateAsync(
+                    assetProvider, solutionChecksum, cancellationToken).ConfigureAwait(false);
 
-                    // get new solution info and options
-                    var solutionInfo = await assetProvider.CreateSolutionInfoAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
-                    return CreateSolutionFromInfo(solutionInfo);
-                }
+                // Now, bring that solution in line with the snapshot defined by solutionChecksum.
+                var updater = new SolutionCreator(Services.HostServices, assetProvider, solutionToUpdate);
+                return await updater.CreateSolutionAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
```

Comment:
once we get the solution to get update, now we actually do the work to ensure it is in sync with teh host side.